### PR TITLE
Refactor/bgp peer

### PR DESCRIPTION
**Pull request by maksym-nazarenko**

### CODE REVIEW

File: client/bgp_peer.go [hunk 1]
```diff
@@ -1,135 +1,112 @@
...
-	RemoteAddress        string `mikrotik:"remote-address"`
-	RemoteAs             int    `mikrotik:"remote-as"`
-	RemotePort           int    `mikrotik:"remote-port"`
-	RemovePrivateAs      bool   `mikrotik:"remove-private-as"`
-	RouteReflect         bool   `mikrotik:"route-reflect"`
-	TCPMd5Key            string `mikrotik:"tcp-md5-key"`
-	TTL                  string `mikrotik:"ttl"`
-	UpdateSource         string `mikrotik:"update-source"`
-	UseBfd               bool   `mikrotik:"use-bfd"`
+	Id                   string `mikrotik:".id" codegen:"id,mikrotikID"`
```
Comment (maksym-nazarenko) [inline client/bgp_peer.go:9]: this change also breaks the `client` module

File: mikrotik/resource_generic_crud_operations.go [hunk 2]
```diff
@@ -61,7 +61,11 @@ func GenericReadResource(terraformModel interface{}, mikrotikModel client.Resour
 			return
 		}
 
-		resource, err := client.Find(mikrotikModel)
+		resource, err := mikrotikClient.Find(mikrotikModel)
+		if client.IsNotFoundError(err) {
```
Comment (maksym-nazarenko) [inline mikrotik/resource_generic_crud_operations.go:65]: this check was missing, so it was a bug.
I didn't create a separate pair of issue/PR for it, but let me know if you want it to be split for better visibility.

**Review by ddelnano — APPROVED** [review]
Sorry for the late review. lgtm!

File: mikrotik/resource_generic_crud_operations.go (hunk 2 above)
Comment (ddelnano) [reply mikrotik/resource_generic_crud_operations.go:65]: I think it's ok to fix it inline here.

**Pull request merged**